edge-modules/iotedge-diagnostics-dotnet/src/Program.cs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Diagnostics
+{
+    using System;
+    using System.IO;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Sockets;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Client.Transport.Mqtt;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.Concurrency;
+    using Microsoft.Azure.Devices.Edge.Util.TransientFaultHandling;
+    using Microsoft.Azure.Devices.Shared;
+    using Microsoft.Extensions.Configuration;
+    using Newtonsoft.Json;
+    using ProxyLib.Proxy;
+    using ProxyLib.Proxy.Exceptions;
+
+    class Program
+    {
+        public static int Main(string[] args)
+        {
+            try
+            {
+                MainAsync(args).Wait();
+            }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine(ex.Message);
+                return 1;
+            }
+
+            return 0;
+        }
+
+        static async Task MainAsync(string[] args)
+        {
+            var config = new ConfigurationBuilder().AddCommandLine(args).Build();
+            switch (args[0])
```

Comment:
do we need to check args[0] exists to avoid index out of range exception?

Reply:
And do we need to validate each argument value not null or empty?

Reply:
it'll just error, which is fine. This is called from the iotedge check script, which will always pass a param. I don;t expect customers to be running this manually